src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -681,28 +681,59 @@
      * </pre>
      * <p>parses as:</p>
      * <pre>
-     * METHOD_DEF -&gt; METHOD_DEF
-     *     |--MODIFIERS -&gt; MODIFIERS
-     *     |--TYPE -&gt; TYPE
-     *     |   `--LITERAL_VOID -&gt; void
-     *     |--IDENT -&gt; foo
-     *     |--LPAREN -&gt; (
-     *     |--PARAMETERS -&gt; PARAMETERS
-     *     |   |--PARAMETER_DEF -&gt; PARAMETER_DEF
-     *     |   |   |--MODIFIERS -&gt; MODIFIERS
-     *     |   |   |--TYPE -&gt; TYPE
-     *     |   |   |   `--LITERAL_INT -&gt; int
-     *     |   |   `--IDENT -&gt; firstParameter
-     *     |   |--COMMA -&gt; ,
-     *     |   `--PARAMETER_DEF -&gt; PARAMETER_DEF
-     *     |       |--MODIFIERS -&gt; MODIFIERS
-     *     |       |--TYPE -&gt; TYPE
-     *     |       |   `--LITERAL_INT -&gt; int
-     *     |       |--ELLIPSIS -&gt; ...
-     *     |       `--IDENT -&gt; secondParameter
-     *     |--RPAREN -&gt; )
-     *      `--SLIST -&gt; {
-     *          `--RCURLY -&gt; }
+     * METHOD_DEF -> METHOD_DEF
+     *  |--MODIFIERS -> MODIFIERS
+     *  |   `--LITERAL_PUBLIC -> public
+     *  |--TYPE -> TYPE
+     *  |   `--LITERAL_VOID -> void
+     *  |--IDENT -> sum
+     *  |--LPAREN -> (
+     *  |--PARAMETERS -> PARAMETERS
+     *  |   |--PARAMETER_DEF -> PARAMETER_DEF
+     *  |   |   |--MODIFIERS -> MODIFIERS
+     *  |   |   |--TYPE -> TYPE
+     *  |   |   |   `--LITERAL_INT -> int
+     *  |   |   `--IDENT -> num1
+     *  |   |--COMMA -> ,
+     *  |   `--PARAMETER_DEF -> PARAMETER_DEF
+     *  |       |--MODIFIERS -> MODIFIERS
+     *  |       |--TYPE -> TYPE
+     *  |       |   `--LITERAL_INT -> int
+     *  |       |--ELLIPSIS -> ...
+     *  |       `--IDENT -> nums
+     *  |--RPAREN -> )
+     *  `--SLIST -> {
+     *      |--VARIABLE_DEF -> VARIABLE_DEF
+     *      |   |--MODIFIERS -> MODIFIERS
+     *      |   |--TYPE -> TYPE
+     *      |   |   `--LITERAL_INT -> int
+     *      |   |--IDENT -> sum
+     *      |   `--ASSIGN -> =
+     *      |       `--EXPR -> EXPR
+     *      |           `--IDENT -> num1
+     *      |--SEMI -> ;
+     *      |--LITERAL_FOR -> for
```

Comment:
Please use AST for example that you have in code tag

Reply:
@romani Do you mean that I should update AST along with line number in TokenTypes.java

Reply:
Put in file `void foo(int firstParameter, int... secondParameter) {}`  and copy AST only  for this line

Reply:
Okk I will do it